src/openzaak/components/catalogi/api/viewsets/zaaktype.py

```diff
@@ -172,20 +171,52 @@ def perform_update(self, serializer):
     def publish(self, request, *args, **kwargs):
         instance = self.get_object()
 
+        related_errors = dict()
+
         # check related objects
         if (
             instance.besluittypen.filter(concept=True).exists()
             or instance.informatieobjecttypen.filter(concept=True).exists()
             or instance.deelzaaktypen.filter(concept=True).exists()
         ):
             msg = _("All related resources should be published")
-            raise ValidationError(
-                {api_settings.NON_FIELD_ERRORS_KEY: msg}, code="concept-relation"
+            related_errors[api_settings.NON_FIELD_ERRORS_KEY] = [msg]
+
+        has_invalid_resultaattypen = instance.resultaattypen.filter(
```

Comment:
It has a lot of duplicated code with the admin check. 
I think it should be moved to a separate validator, which will be called both here and in the admin

Reply:
Added to a separate validator